internal/template/mapper/mapper.go

```diff
@@ -108,7 +109,8 @@ func injectDynamicRes(
 	return m, nil
 }
 
-func injectDynamicResString(val string) (string, error) {
+func mapDynamicValuesAndPathsString(val string) (string, error) {
+	// If the value contains dynamic values, we need to parse them.
 	dynamicValues := regexp.MustCompile(`{(.*?)}`).FindAllString(val, -1)
```

Comment:
this can be optimized by moving the MustCompile to a package variable, to avoid the compilation to happen over and over again